### ISSUE TITLE
Windows sync client connection issues after hibernation

### ISSUE THREAD

**Issue by rimeraz** [nextcloud/desktop#324]
## Details

- Windows 10, version 1703 (Build 15063.1088), company laptop, using system proxy
- Sync client 2.3.3 build 1
- Nextcloud 13 von CentOS 7.4 VPS

## JavaScript Console Output

I think this does not apply in this case

## Steps to reproduce

Not clearly reproducible

1. Hibernate Windows
2. Wake up from hibernation
3. Notice gray tray icon and see "No connection to $host: Timeout" in sync client

Clicking the $host in the error message opens a browser window and shows $host's login page. Pausing and resuming the sync does not help.

The only solution so far is to restart the client which immediately reconnects.

**Comment by camilasan** [comment]
Could you provide the client logs?

**Comment by rimeraz** [comment]
This is from `.owncloudsync.log` for last two days:

```
[... snip ...]
#=#=#=# Syncrun started 2018-05-16T12:41:55
#=#=#=#=# Propagation starts 2018-05-16T12:41:56 (last step: 451 msec, total: 451 msec)
#=#=#=# Syncrun finished 2018-05-16T12:41:56 (last step: 36 msec, total: 488 msec)
#=#=#=# Syncrun started 2018-05-16T15:10:56
#=#=#=#=# Propagation starts 2018-05-16T15:11:06 (last step: 10525 msec, total: 10525 msec)
#=#=#=# Syncrun finished 2018-05-16T15:11:06 (last step: 20 msec, total: 10545 msec)
#=#=#=# Syncrun started 2018-05-16T15:11:12
#=#=#=# Syncrun started 2018-05-16T15:14:26
#=#=#=#=# Propagation starts 2018-05-16T15:14:29 (last step: 2816 msec, total: 2816 msec)
#=#=#=# Syncrun finished 2018-05-16T15:14:29 (last step: 17 msec, total: 2834 msec)
#=#=#=# Syncrun started 2018-05-17T07:41:56
#=#=#=#=# Propagation starts 2018-05-17T07:42:11 (last step: 15132 msec, total: 15132 msec)
#=#=#=# Syncrun finished 2018-05-17T07:42:11 (last step: 28 msec, total: 15160 msec)
#=#=#=# Syncrun started 2018-05-17T08:30:18
#=#=#=#=# Propagation starts 2018-05-17T08:30:19 (last step: 425 msec, total: 425 msec)
#=#=#=# Syncrun finished 2018-05-17T08:30:19 (last step: 29 msec, total: 454 msec)
#=#=#=# Syncrun started 2018-05-17T10:30:20
#=#=#=#=# Propagation starts 2018-05-17T10:30:20 (last step: 497 msec, total: 497 msec)
#=#=#=# Syncrun finished 2018-05-17T10:30:20 (last step: 26 msec, total: 524 msec)
#=#=#=# Syncrun started 2018-05-17T12:30:22
#=#=#=#=# Propagation starts 2018-05-17T12:30:23 (last step: 510 msec, total: 510 msec)
#=#=#=# Syncrun finished 2018-05-17T12:30:23 (last step: 28 msec, total: 539 msec)
```

Laptop has been hibernated 2018-05-16 17:00 and I noticed the issue 2018-05-17 07:40.

Maybe it is related to the syncrun started 2018-05-16 15:11:12 which has not been finished. But IIRC the laptop needed a reboot at that time.

I will try to provide logs from within the sync client via F12 when the issue occurs again.

**Comment by haegelix** [comment]
same problem here.

**Comment by en00x** [comment]
+1
2.5.1. Windows 10

**Comment by Chingato** [comment]
I have the same problem with Windows 10 Home, version 1809, build 17763.316, OS language: German

But I not have a gray icon, but a red cross. If I close the desktop client and start it again it works. But this requires manual intervention after each wake up from hibernate which should not be the case.

Client Log:
|[OCC::ActivityListModel::startFetchJob |Start fetching activities for "Thomas@192.168.253.23"|
|—|---|
|[OCC::AccessManager::createRequest |2 “” “http://192.168.253.23/cloud/ocs/v2.php/cloud/activity?start=0&count=100&format=json” has X-Request-ID “0b4002c0-35d7-4744-9fe0-5d8814b7566c”|
|[OCC::AbstractNetworkJob::start |OCC::JsonApiJob created for “http://192.168.253.23/cloud” + “ocs/v2.php/cloud/activity” “OCC::ActivityListModel”|
|[OCC::AccessManager::createRequest |2 “” “http://192.168.253.23/cloud/ocs/v2.php/apps/notifications/api/v2/notifications?format=json” has X-Request-ID “e6a22fe3-c10d-4542-a09e-40a813891721”|
|[OCC::AbstractNetworkJob::start |OCC::JsonApiJob created for “http://192.168.253.23/cloud” + “ocs/v2.php/apps/notifications/api/v2/notifications” “OCC::ServerNotificationHandler”|
|[OCC::WebFlowCredentials::slotFinished |request finished|
|[OCC::AbstractNetworkJob::slotFinished |QNetworkReply::NetworkError(UnknownNetworkError) “Der Zugriff auf das Netzwerk ist nicht gestattet.” QVariant(Invalid)|
|[OCC::WebFlowCredentials::stillValid |Still valid?|
|[OCC::WebFlowCredentials::stillValid |QNetworkReply::NetworkError(UnknownNetworkError)|
|[OCC::WebFlowCredentials::stillValid |“Der Zugriff auf das Netzwerk ist nicht gestattet.”|
|[OCC::JsonApiJob::finished |JsonApiJob of QUrl(“http://192.168.253.23/cloud/ocs/v2.php/cloud/activity?start=0&count=100&format=json”) FINISHED WITH STATUS “UnknownNetworkError Der Zugriff auf das Netzwerk ist nicht gestattet.”|
|[OCC::JsonApiJob::finished |Network error: “ocs/v2.php/cloud/activity” “Der Zugriff auf das Netzwerk ist nicht gestattet.” QVariant(Invalid)|
|[OCC::WebFlowCredentials::slotFinished |request finished|
|[OCC::AbstractNetworkJob::slotFinished |QNetworkReply::NetworkError(UnknownNetworkError) “Der Zugriff auf das Netzwerk ist nicht gestattet.” QVariant(Invalid)|
|[OCC::WebFlowCredentials::stillValid |Still valid?|
|[OCC::WebFlowCredentials::stillValid |QNetworkReply::NetworkError(UnknownNetworkError)|
|[OCC::WebFlowCredentials::stillValid |“Der Zugriff auf das Netzwerk ist nicht gestattet.”|
|[OCC::JsonApiJob::finished |JsonApiJob of QUrl(“http://192.168.253.23/cloud/ocs/v2.php/apps/notifications/api/v2/notifications?format=json”) FINISHED WITH STATUS “UnknownNetworkError Der Zugriff auf das Netzwerk ist nicht gestattet.”|
|[OCC::JsonApiJob::finished |Network error: “ocs/v2.php/apps/notifications/api/v2/notifications” “Der Zugriff auf das Netzwerk ist nicht gestattet.” QVariant(Invalid)|
|[OCC::ServerNotificationHandler::slotNotificationsReceived |Notifications failed with status code 0|
|[OCC::Folder::slotRunEtagJob |Trying to check “http://192.168.253.23/cloud/remote.php/dav/files/Thomas/ 1” for changes via ETag check. (time since last sync: 500 s)|
|[OCC::AccessManager::createRequest |6 “PROPFIND” “http://192.168.253.23/cloud/remote.php/dav/files/Thomas/ 1” has X-Request-ID “7ece8acf-9c9e-42fa-8f7c-6bd4a67a6ec4”|
|[OCC::RequestEtagJob::start |request network error: “Der Zugriff auf das Netzwerk ist nicht gestattet.”|
|[OCC::AbstractNetworkJob::start |OCC::RequestEtagJob created for “http://192.168.253.23/cloud” + “/” “OCC::Folder”|
|[OCC::WebFlowCredentials::slotFinished |request finished|
|[OCC::AbstractNetworkJob::slotFinished |QNetworkReply::NetworkError(UnknownNetworkError) “Der Zugriff auf das Netzwerk ist nicht gestattet.” QVariant(Invalid)|
|[OCC::WebFlowCredentials::stillValid |Still valid?|
|[OCC::WebFlowCredentials::stillValid |QNetworkReply::NetworkError(UnknownNetworkError)|
|[OCC::WebFlowCredentials::stillValid |“Der Zugriff auf das Netzwerk ist nicht gestattet.”|
|[OCC::RequestEtagJob::finished |Request Etag of QUrl(“http://192.168.253.23/cloud/remote.php/dav/files/Thomas/ 1”) FINISHED WITH STATUS “UnknownNetworkError Der Zugriff auf das Netzwerk ist nicht gestattet.”|
|[OCC::ActivityWidget::slotPrimaryButtonClickedOnListView |Opening “d:%5CNextcloud%5Clocal_sync_Thomas” in browser for Notification/Activity “Beim Öffnen eines Ordners ist ein Fehler aufgetreten. Der Zugriff auf das Netzwerk ist nicht gestattet.”|
|[unknown |ShellExecute ‘d:%5CNextcloud%5Clocal_sync_Thomas’ failed (error 2).|
|[OCC::Utility::openBrowser |QDesktopServices::openUrl failed for QUrl(“d:%5CNextcloud%5Clocal_sync_Thomas”)|

**Comment by panspectron2017** [comment]
This is really annoying on laptops and big repos, as I often change VPN and often send the laptop to sleep. 
The repos practically never get synced, as the sync-client is always offline.

**Comment by darren-harrison** [comment]
I have been monitoring this thread, hoping for a quick fix. Our company has a number of people with the same problem.

**Comment by swaechter** [comment]
I am suffering from the same problem. I work part time from home, so to often data "stays at home" ;)

**Comment by ThomasCr** [comment]
I am have also this problem for longer time. Only a restart of the app helps ... thats annoying.
I also have a red cross. The context menu opens also just fine, but if I double click or also try rightclick > preferences the window can't open. Only a Nextcloud entry is on the Taskbar - but no window pops up. When I hover over the entry I have only a blank window in the preview.

![grafik](https://user-images.githubusercontent.com/226670/56151173-f58c3480-5fb0-11e9-83ea-2971f5e97652.png)
![grafik](https://user-images.githubusercontent.com/226670/56151249-266c6980-5fb1-11e9-82aa-66f222465ba2.png)

**Comment by Chingato** [comment]
for me the issue is solved since I upgraded the desktop sync client. I am using hibernate under Windows 10 (1809)

Server: 15.0.7
Desktop: 2.5.2git (build 20190319)

**Comment by swaechter** [comment]
I can confirm what @Chingato reported. I have three different servers/accounts and 2.5.2 resolved the problem

**Comment by nagua** [comment]
I still have the problem with 2.6.4.

**Comment by rimeraz** [comment]
I did not follow up on this because after some point the issue did not appear anymore.

But now I have it again on my home PC. This time with the white cross in red circle as mentioned by the other folks here.

OS: Windows 10 20H2 fresh install (3 days old). It appeared from the beginning.
Nextcloud client: 3.1.1

Steps to reproduce explained from the beginning are still valid.

The only solution so far is client restart. Pausing sync and forcing does not solve it.

Nextcloud_sync.log:

```
#=#=#=# Syncrun started 2021-01-12T20:39:15Z
#=#=#=#=# Propagation starts 2021-01-12T20:39:15Z (last step: 230 msec, total: 230 msec)
#=#=#=# Syncrun finished 2021-01-12T20:39:15Z (last step: 11 msec, total: 242 msec)
#=#=#=# Syncrun started 2021-01-12T22:39:17Z
#=#=#=#=# Propagation starts 2021-01-12T22:39:18Z (last step: 199 msec, total: 199 msec)
#=#=#=# Syncrun finished 2021-01-12T22:39:18Z (last step: 8 msec, total: 208 msec)
#=#=#=# Syncrun started 2021-01-13T06:35:20Z
#=#=#=# Syncrun finished 2021-01-13T06:35:20Z (last step: 109 msec, total: 109 msec)
#=#=#=# Syncrun started 2021-01-13T06:35:34Z
#=#=#=# Syncrun finished 2021-01-13T06:35:34Z (last step: 87 msec, total: 87 msec)
#=#=#=# Syncrun started 2021-01-13T06:36:41Z
#=#=#=# Syncrun finished 2021-01-13T06:36:41Z (last step: 96 msec, total: 96 msec)
```

owncloud.log:
[20210113_0736_owncloud.log](https://github.com/nextcloud/desktop/files/5807090/20210113_0736_owncloud.log)

The error message in my case is in German _UnknownNetworkError Der Zugriff auf das Netzwerk ist nicht gestattet._ and would be something like "Accessing the network is not permitted." or "not allowed".

**Comment by rimeraz** [comment]
Workaround without restarting the client: Log off user and log on again in main window. Might require grant access in Nextcloud instance in browser window

**Comment by tomdereub** [comment]
Same problem for me, I started to have the problem with nextcloud client v 3.1.1. Downgrading to 3.1.0 seems to solve the problem.

**Comment by rimeraz** [comment]
for me 3.1.0 behaves the same way. So downgrading does not help me until the fixed version has been released.

**Comment by sergio-ns** [comment]
I have the same problem, Windows 10 version 3.1.1. Another workaround is to change proxy settings, e.g. from "No proxy" to "Use system" and back to "No proxy". Very annoying.

**Comment by Chingato** [comment]
does anyone know how the process is to get this issue re-opened? Currently it is closed and I have some doubts that it will be fixed if we just add comments here but the issue itself remains closed.

@er-vin any idea?

**Comment by FlexW** [comment]
This issue should be resolved with https://github.com/nextcloud/client-building/pull/63

**Comment by er-vin** [comment]
As @FlexW said, this has been closed because a fix was produced. To be shipped with 3.1.2.
